### DISCUSSION TITLE
Does renovate support the standard Gradle Version Catalog?

### DISCUSSION THREAD

**Discussion by hpuac** [renovatebot/renovate#26231] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.87.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Hey folks, do you actually support parsing the regular Gradle Version Catalog?
I saw the documentation mentioning toml file but I'm wondering if non toml versions are also supported.
We have our versions in the `buildSrc/settings.gradle.kts` and the file is picked up but the error we get is:
```
DEBUG: Failed to process Gradle file (repository=boo/bar, packageFile=buildSrc/settings.gradle.kts)
       "err": {
         "message": "Expected token version not found",
         "stack": "Error: Expected token version not found\n at loadFromTokenMap (/opt/containerbase/tools/renovate/37.43.3/node_modules/renovate/lib/modules/manager/gradle/parser/common.ts:72:11)\n [...]
```

Our `buildSrc/settings.gradle.kts` looks like this (just a demo):
```
rootProject.name = "buildSrc"

dependencyResolutionManagement {
    versionCatalogs {
        create("catalog") {
            version("kotlin", "1.9.20").also {
                plugin("kotlin-jvm", "org.jetbrains.kotlin.jvm").versionRef(it)
                plugin("kotlin-kapt", "org.jetbrains.kotlin.kapt").versionRef(it)
                plugin("kotlin-multiplatform", "org.jetbrains.kotlin.multiplatform").versionRef(it)

                plugin("kotlin-plugin-allopen", "org.jetbrains.kotlin.plugin.allopen").versionRef(it)
                plugin("kotlin-plugin-serialization", "org.jetbrains.kotlin.plugin.serialization").versionRef(it)
                plugin("kotlin-plugin-spring", "org.jetbrains.kotlin.plugin.spring").versionRef(it)

                library("kotlin-reflect", "org.jetbrains.kotlin", "kotlin-reflect").versionRef(it)
            }

            version("spring-boot", "3.1.5").also {
                plugin("spring-boot", "org.springframework.boot").versionRef(it)
            }

            plugin("checksum", "org.gradle.crypto.checksum").version("1.4.0")                        
            plugin("download", "de.undercouch.download").version("5.3.0")                            

            library("apache-commons-text", "org.apache.commons:commons-text:1.9")
            library("apache-httpComponents-httpclient", "org.apache.httpcomponents:httpclient:4.5.13")
            [...]
        }
    }
}
```

Do we do something wrong or is it simply not supported (yet).

Cheers
Hans

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by Churro** [comment]
Can you please provide a minimal reproduction repo?

Renovate handles regular gradle version catalogs well but a particular construct you are using (most likely `version("kotlin", "1.9.20").also {`) may need an additional pattern that is not yet considered by the parser.

**Reply by hpuac** [reply]
Thank you for the quick reply! I will prepare something and will get back to you, thx!

**Reply by hpuac** [reply]
Hey @Churro , I created a demo repository here: https://github.com/hpuac/renovate-gradle-version-catalog-demo
Let me know if you need anything else!

**Reply by Churro** [reply, marked as answer]
Thanks, that helped! The gradle manager in renovate currently doesn't support Kotlin scope functions and their use with inlined version catalogs doesn't seem to be very common. In any case I intend to address this because it's a legit pattern and the error shouldn't be shown.

**Reply by mkurde** [reply]
As followup question to your answer: Are you working on addressing this in the near future?

**Reply by Churro** [reply]
I've just created PR #26737 to address the issue that stops renovate from being able to handle plugins in version catalogs that are immediately followed by a version. That allows it to run successfully on the reproduction repo, see https://github.com/renovate-demo/26231-renovate-gradle-version-catalog-demo/pulls
